docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,734 @@
+     - :class:`~qiskit.quantum_info.StabilizerState` or :class:`~qiskit.quantum_info.Statevector` or :class:`~qiskit.QuantumCircuit`, depending on the use case
+     - In principle, :class:`~qiskit.quantum_info.StabilizerState` is the most efficient replacement for :class:`~qiskit.opflow` states, but the functionality is not identical. See API ref. for more info.
+
+   * -
+
+        .. code-block:: python
+
+            from qiskit.opflow import Zero, One, Plus, Minus
+
+            state1 = Zero ^ One
+            state2 = Plus ^ Minus
+
+     -
+
+        .. code-block:: python
+
+            from qiskit import QuantumCircuit
+            from qiskit.quantum_info import StabilizerState
+
+            qc_zero = QuantumCircuit(1)
+            qc_one = copy(qc_zero)
+            qc_one.x(0)
+            state1 = StabilizerState(qc_zero) ^ StabilizerState(qc_one)
+
+            qc_plus = copy(qc_zero)
+            qc_plus.h(0)
+            qc_minus = copy(qc_one)
+            qc_minus.h(0)
+            state2 = StabilizerState(qc_plus) ^ StabilizerState(qc_minus)
+     -
+
+
+
+Primitive and List Ops
+----------------------
+Most of the workflows that previously relied in components from :mod:`~qiskit.opflow.primitive_ops` and :mod:`~qiskit.opflow.list_ops` can now
+leverage elements from :mod:`~qiskit.quantum_info.operators` instead. Some of these classes do not require a 1-1 replacement because
+they were created to interface with other opflow components.
+
+PrimitiveOps
+~~~~~~~~~~~~~~
+
+.. list-table:: Migration of ``qiskit.opflow.primitive_ops``
+   :header-rows: 1
+
+   * - opflow
+     - alternative
+     - notes
+
+   * - :class:`~qiskit.opflow.PrimitiveOp`
+     - No direct replacement
+     - Can directly use :class:`~qiskit.quantum_info.Operator``
+   * - :class:`~qiskit.opflow.CircuitOp`
+     - No direct replacement
+     - Can directly use :class:`~qiskit.QuantumCircuit`
+   * - :class:`~qiskit.opflow.MatrixOp`
+     - :class:`~qiskit.quantum_info.Operator``
+     -
+   * - :class:`~qiskit.opflow.PauliOp`
+     - :class:`~qiskit.quantum_info.Pauli`
+     - For direct compatibility with classes in :mod:`~qiskit.algorithms`, wrap in :class:`~qiskit.quantum_info.SparsePauliOp`
+   * - :class:`~qiskit.opflow.PauliSumOp`
+     - :class:`~qiskit.quantum_info.SparsePauliOp`
+     - See example below
+   * - :class:`~qiskit.opflow.TaperedPauliSumOp`
+     - This class was used to combine a :class:`~PauliSumOp` with its identified symmetries in one object. It has been deprecated without replacement
+     - See ``Z2Symmetries`` example for updated workflow
+   * - :class:`~qiskit.opflow.Z2Symmetries`
+     - :class:`~qiskit.quantum_info.Z2Symmetries`
+     - See example below
+
+
+PrimitiveOps Examples
+~~~~~~~~~~~~~~~~~~~~~
+.. list-table::
+   :header-rows: 1
+
+   * - opflow
+     - alternative
+     - notes
+
+   * -  ``PauliSumOp`` **Example:**
+
+        .. code-block:: python
+
+            from qiskit.opflow import PuliSumOp
```

Comment:
@mtreinish?

Reply:
So @woodsp-ibm, I managed to find an inbetween solution, I formatted the code examples to be runnable locally with pytest's `doctest`. I don't know if I like having all of the `>>>` in the code, but at least we can make sure that it runs.

Reply:
It is also possible to use ..testcode:: and ..testoutput:: directives (they are supported by the sphinx.ext.doctest, which we use in the apps and imagine would be used here too). By default the output puts these in separate highlighted blocks (though you can hide the testoutput if desired) but there is no need for >>> when using these.

I know though we used >>> in Nature etc since formatting with these former seemed harder to get things to come out right/as wanted etc. The >>> was always more concise and seemed easier to add and have it come out as expected in among other text. Maybe there is styling that can applied to testcode/output so things can be improved, not sure.

E.g.

```
.. testcode::

    x = 1 + 3
    print(x)

.. testoutput::

    4

```
comes out as
![image](https://user-images.githubusercontent.com/40241007/222256752-15e0a92f-04b8-46b9-9e2f-520cc2fc6ae7.png)